File: backend/controllers/servicesController.js
const db = require('../config/db');

exports.getServices = async (req, res, next) => {
    try {

        const servicesResult = await db.query(
            'SELECT s.id, s.name, MIN(o.rate) minrate FROM services s, offerings o WHERE o.service_id = s.id GROUP BY s.id'
        );

        let services = [];
        servicesResult.rows.forEach(e => {
            services.push({service_id: e.id, name: e.name, minRate: e.minrate});
        });

        return res.json(services);
    } catch (err) {
        next(err);
    }
}

File: backend-csharp/Controllers/ServicesController.cs
using HomeServicesPlatform.Data;
using HomeServicesPlatform.Models;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace HomeServicesPlatform.Controllers
{
    [ApiController]
    [Route("api/v1/services")]
    public class ServicesController : ControllerBase
    {
        private readonly ApplicationDbContext _context;

        public ServicesController(ApplicationDbContext context)
        {
            _context = context;
        }

        [HttpGet]
        public async Task<IActionResult> GetServices()
        {
            var services = await _context.services
                .Join(_context.offerings,
                    s => s.id,
                    o => o.service_id,
                    (s, o) => new { s, o })
                .GroupBy(x => x.s.id)
                .Select(g => new
                {
                    service_id = g.First().s.id,
                    name = g.First().s.name,
                    minRate = g.Min(x => x.o.rate)
                })
                .ToListAsync();

            return Ok(services);
        }
    }
}
